### PR TITLE
Notifications filtered by projects

### PR DESCRIPTION
**Pull request by saraycp**
This PR is a follow-up to #9433.

The notifications can be now filtered by the projects associated to them.
To do so, a list of project names and a the number of notifications per project are displayed.

Screenshot_2020-04-25 Open Build Service

Co-authored-by: Eduardo Navarro <enavarro@suse.com>

### CODE REVIEW

File: src/api/app/controllers/webui/users/notifications_controller.rb [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+    @notifications = if params[:project]
+                       NotificationsFinder.new(notifications_for_subscribed_user).for_project_name(params[:project])
+                     else
+                       NotificationsFinder.new(notifications_for_subscribed_user).for_notifiable_type(params[:type])
+                     end
+
+    @projects = Project.joins(:notifications)
+                       .where(notifications: { subscriber: User.session, delivered: false })
+                       .order('name desc').group(:name).count
+                       .sort_by(&:last).reverse.to_h
```
Comment (vpereira) [inline src/api/app/controllers/webui/users/notifications_controller.rb:19]: wouldn't be better to just call `reverse.to_h` and fire the query when it's actually needed and have `@projects` as actually an activerecord collection?

Comment (danidoni) [inline src/api/app/controllers/webui/users/notifications_controller.rb:19]: My only improvement is to move this four lines into a method with a descriptive name so we can know at a quick glance what this is doing.

Comment (saraycp) [reply src/api/app/controllers/webui/users/notifications_controller.rb]: @vpereira this has been addressed as discussed offline. Thanks for your suggestion! :bouquet:

Comment (saraycp) [reply src/api/app/controllers/webui/users/notifications_controller.rb]: @danidoni thanks! It's been moved to the method `projects_for_filter`.

**Comment by saraycp** [comment]
I had to do some little changes when rebasing. Could you have another look? @vpereira and @danidoni 

Pay special attention to [this file](https://github.com/openSUSE/open-build-service/pull/9447/files#diff-3fea9b194ee535ea7887130c45fc9988R59) and  [this method](https://github.com/openSUSE/open-build-service/pull/9447/files#diff-3fea9b194ee535ea7887130c45fc9988R59).

**Pull request merged**